src/main/java/com/spotify/reaper/resources/view/NodesStatus.java

```diff
@@ -0,0 +1,208 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.spotify.reaper.resources.view;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Optional;
+
+import jersey.repackaged.com.google.common.collect.Lists;
+import jersey.repackaged.com.google.common.collect.Maps;
+
+import java.math.BigDecimal;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.regex.*;
+import java.util.stream.Collectors;
+
+public class NodesStatus {
+  @JsonProperty
+  public final List<GossipInfo> endpointStates;
+  private static Pattern endpointNamePattern = Pattern.compile("^([0-9]{1,3}\\.[0-9]{1,3}\\.[0-9]{1,3}\\.[0-9]{1,3})", Pattern.MULTILINE);
+  private static Pattern endpointStatusPattern = Pattern.compile("(STATUS):([0-9]*):(\\w+)");
+  private static Pattern endpointDcPattern = Pattern.compile("(DC):([0-9]*):(\\w+)");
+  private static Pattern endpointRackPattern = Pattern.compile("(RACK):([0-9]*):(\\w+)");
+  private static Pattern endpointLoadPattern = Pattern.compile("(LOAD):([0-9]*):([0-9eE.]+)");
+  private static Pattern endpointReleasePattern = Pattern.compile("(RELEASE_VERSION):([0-9]*):([0-9.]+)");
+  private static Pattern endpointSeverityPattern = Pattern.compile("(SEVERITY):([0-9]*):([0-9.]+)");
+  private static Pattern endpointHostIdPattern = Pattern.compile("(HOST_ID):([0-9]*):([0-9a-z-]+)");
+  private static Pattern endpointTokensPattern = Pattern.compile("(TOKENS):([0-9]*)");
+  private Matcher matcher;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the "matcher" field and declare it as a local variable in the relevant methods. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1450)